### PR TITLE
Cleanup fixed hash

### PR DESCRIPTION
**Pull request by chriseth**
Depends on https://github.com/ethereum/solidity/pull/2808

### CODE REVIEW

File: libdevcore/FixedHash.h [hunk 1]
```diff
@@ -97,33 +89,14 @@ class FixedHash
 	bool operator==(FixedHash const& _c) const { return m_data == _c.m_data; }
 	bool operator!=(FixedHash const& _c) const { return m_data != _c.m_data; }
 	bool operator<(FixedHash const& _c) const { for (unsigned i = 0; i < N; ++i) if (m_data[i] < _c.m_data[i]) return true; else if (m_data[i] > _c.m_data[i]) return false; return false; }
```
Comment (axic) [inline libdevcore/FixedHash.h]: Please add the comment that this is needed for map.

File: libdevcore/FixedHash.h [hunk 2]
```diff
@@ -97,33 +89,14 @@ class FixedHash
...
 	/// @returns a particular byte from the hash.
 	byte& operator[](unsigned _i) { return m_data[_i]; }
 	/// @returns a particular byte from the hash.
 	byte operator[](unsigned _i) const { return m_data[_i]; }
 
-	/// @returns an abridged version of the hash as a user-readable hex string.
-	std::string abridged() const { return toHex(ref().cropped(0, 4)) + "\342\200\246"; }
-
 	/// @returns a version of the hash as a user-readable hex string that leaves out the middle part.
 	std::string abridgedMiddle() const { return toHex(ref().cropped(0, 4)) + "\342\200\246" + toHex(ref().cropped(N - 4)); }
```
Comment (axic) [inline libdevcore/FixedHash.h]: What is the point of this?

Comment (chriseth) [reply libdevcore/FixedHash.h]: This is used in the AST printer to shorten a hash.

Comment (axic) [reply libdevcore/FixedHash.h]: We can consider replacing that without a shorted version in a different PR then.

**Review by axic — APPROVED** [review]

**Pull request merged**